src/screens/Departures/FavouriteStopsOverview.tsx

```diff
@@ -0,0 +1,88 @@
+import React from 'react';
+import {useStopsDetailsData} from '@atb/screens/Departures/state/stop-place-details-state';
+import {useFavorites} from '@atb/favorites';
+import {Place, StopPlacePosition} from '@atb/api/types/departures';
+import {CompositeNavigationProp} from '@react-navigation/native';
+import DeparturesTexts from '@atb/translations/screens/Departures';
+import {useTranslation} from '@atb/translations';
+import StopPlaces, {
+  NoStopPlaceMessage,
+} from '@atb/screens/Departures/components/StopPlaces';
+import {StackNavigationProp} from '@react-navigation/stack';
+import {DeparturesStackParams} from '@atb/screens/Departures/index';
+import {RootStackParamList} from '@atb/navigation';
+import {useGeolocationState} from '@atb/GeolocationContext';
+import {coordinatesDistanceInMetres} from '@atb/utils/location';
+
+type FavouritePlacesNavigationProp = CompositeNavigationProp<
+  StackNavigationProp<DeparturesStackParams>,
+  StackNavigationProp<RootStackParamList>
+>;
+
+type RootProps = {
+  navigation: FavouritePlacesNavigationProp;
+};
+
+const FavouriteStopsOverview = ({navigation}: RootProps) => {
+  const {favoriteDepartures} = useFavorites();
+  const {location} = useGeolocationState();
+  const favouriteStopIds = new Set(favoriteDepartures.map((fd) => fd.stopId));
+  const {state} = useStopsDetailsData(Array.from(favouriteStopIds));
+  const currentLocation = location || undefined;
+
+  const getDistanceFromCurrentLocation = (
+    latitude?: number,
+    longitude?: number,
+  ) => {
+    return (
+      currentLocation &&
+      latitude &&
+      longitude &&
+      coordinatesDistanceInMetres(currentLocation.coordinates, {
+        latitude: latitude,
+        longitude: longitude,
+      })
+    );
+  };
+
+  const favouriteStopsDetails = state.data?.stopPlaces
```

Comment:
This might be something that could be wrapped either in a custom hook or just a useMemo directly. If there are a lot of calculations and a lot of renders.